sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -321,6 +320,21 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
     }
   }
 
+  object DefaultJoin extends Strategy {
```

Comment:
Add the new rule object, for the real last gate, for now, it's actually will generate the `BroadcastNestedLoopJoin` operator, as what we have today, probably we can change that in the future.